### PR TITLE
Fix passing Mac Catalyst -target flag to .mm builds

### PR DESCRIPTION
**Pull request by filipnavara**
Not very useful on its own but it will be necessary for NativeAOT test builds where the ObjectiveC tests uses .mm files.

### CODE REVIEW

File: eng/native/configurecompiler.cmake [hunk 1]
```diff
@@ -549,6 +549,7 @@ if (CLR_CMAKE_HOST_UNIX)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${MACOS_VERSION_MIN_FLAGS} ${DISABLE_OVERRIDING_MIN_VERSION_ERROR}")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${MACOS_VERSION_MIN_FLAGS} ${DISABLE_OVERRIDING_MIN_VERSION_ERROR}")
     set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${MACOS_VERSION_MIN_FLAGS} ${DISABLE_OVERRIDING_MIN_VERSION_ERROR}")
+    set(CMAKE_OBJCXX_FLAGS "${CMAKE_C_FLAGS} ${MACOS_VERSION_MIN_FLAGS} ${DISABLE_OVERRIDING_MIN_VERSION_ERROR}")
```
Comment (akoeplinger) [inline eng/native/configurecompiler.cmake:552]: shouldn't this be `CMAKE_OBJC_FLAGS` ? we're not using Objective-C++ as far as I know

Comment (akoeplinger) [reply eng/native/configurecompiler.cmake:552]: Ah sorry, I just saw this is specifically for .mm files so those _are_ Objective-C++.

Though that raises the question if we should use `CMAKE_C_FLAGS` for `CMAKE_OBJC_FLAGS` and `CMAKE_CXX_FLAGS` for `CMAKE_OBJCXX_FLAGS`

Comment (filipnavara) [reply eng/native/configurecompiler.cmake:552]: Ufff. That was cut & paste error... it should start with `set(CMAKE_OBJCXX_FLAGS "${CMAKE_OBJCXX_FLAGS}`

Comment (filipnavara) [inline eng/native/configurecompiler.cmake:552]:
```suggestion
    set(CMAKE_OBJCXX_FLAGS "${CMAKE_OBJCXX_FLAGS} ${MACOS_VERSION_MIN_FLAGS} ${DISABLE_OVERRIDING_MIN_VERSION_ERROR}")
```

Comment (akoeplinger) [reply eng/native/configurecompiler.cmake]: ok, and what about CMAKE_OBJC_FLAGS?

Comment (filipnavara) [reply eng/native/configurecompiler.cmake]: We can add that too, I guess.

Comment (filipnavara) [reply eng/native/configurecompiler.cmake]: Done

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**